libtenzir/src/detail/series_builders.cpp

```diff
@@ -586,6 +586,12 @@ auto record_series_builder_base::append() -> void {
     std::visit(
       []<class Builder>(Builder& b) {
         if constexpr (std::is_base_of_v<record_series_builder_base, Builder>) {
+          // FIXME: The following line prevents the assertion above from
+          // triggering in the recursive call below. This can happen when a
+          // list<record> builder gets an entry that contains a new field.
+          // Unfortunately adding this line is not sufficient or not the correct
+          // fix, as the data won't be present in the resulting table slice.
```

Comment:
Some textual nits:

* The following line -> The call to `get_arrow_builder()`
* This can happen -> This happens
* not sufficient or not the correct fix -> not sufficient and not the correct fix

Reply:
Let's ignore this unless the current CI run fails, since it's not important enough to run for another hour.